zmon_worker_monitor/builtins/plugins/datapipeline.py

```diff
@@ -62,11 +90,25 @@ def get_details(self, pipeline_ids):
         """
         if isinstance(pipeline_ids, str):
             pipeline_ids = [pipeline_ids]
-        else:
-            if not isinstance(pipeline_ids, list):
+        elif not isinstance(pipeline_ids, list):
                 raise CheckError('Parameter \"pipeline_ids\" should be a string or a list of strings '
                                  'denoting pipeline IDs')
 
+        if isinstance(pipeline_names, str):
+            pipeline_names = [pipeline_names]
+        elif not isinstance(pipeline_names, list):
+                raise CheckError('Parameter \"pipeline_names\" should be a string or a list of strings '
+                                 'denoting pipeline names')
+
+        if not pipeline_ids and not pipeline_names:
```

Comment:
I think this part is duplicated by ``describe_all_pipelines``. So I would remove it from here, as being explicit is *usually* better than implicit.